lib/pbench/server/api/resources/users_api.py

```diff
@@ -8,12 +9,93 @@
 import jwt
 from sqlalchemy.exc import IntegrityError, SQLAlchemyError
 
+from pbench.server import PbenchServerConfig
 from pbench.server.api.auth import Auth
 from pbench.server.database.models.active_tokens import ActiveTokens
 from pbench.server.database.models.server_config import ServerConfig
 from pbench.server.database.models.users import User
 
 
+class AuthenticationSchema(Resource):
+    """
+    Pbench API to return authentication parameters required for the client to
+    authenticate the user with OIDC broker.
+    """
+
+    AVAILABALE_SCHEMA = ["prod", "dev", "test"]
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        self.server_config = config
+        self.logger = logger
+
+    def get(self):
+        """
+        Returns the authentication schema parameters.
+
+        Client can request a specific authorization schema paramters required
+        to authenticate an end-user. However, if the schema is not provided we
+        return the production schema to carry out the user authentication.
+
+        Currently we only support the authentication schemas named mentioned in
+        AVAILABLE_SCHEMA.
+
+        Schema name is required to be a URI parameter. E.g.
+        GET pbench_server_rest_uri/authentication?schema=[schema_name]"
+        :return: JSON Payload
+            Success: 200,
+            response_object = {
+                "schema": [schema-name],
+                "auth_realm": [realm-name],
+                "auth_client": [client-name],
+                "auth_issuer": [auth-issuer],
+                "auth_secret": [secret-string]
+                # empty auth secret means the auth client does not
+                # need a secret to communicate with an auth issuer
+            }
+            Failure: <status_Code>,
+            response_object = {
+                "message": "failure message"
+            }
+        """
+        disabled = ServerConfig.get_disabled()
+        if disabled:
+            abort(HTTPStatus.SERVICE_UNAVAILABLE, **disabled)
+
+        schema = request.args.get("schema")
+        if not schema:
+            schema = "prod"
+        elif schema not in self.AVAILABALE_SCHEMA:
+            abort(
+                HTTPStatus.BAD_REQUEST,
+                message=f"Schema not supoorted; Available schemas: {self.AVAILABALE_SCHEMA}",
+            )
+
+        auth_client = self.server_config.get("authentication", "client")
+        auth_realm = (
+            self.server_config.get("authentication", "realm_prefix")
+        ) + f"_{schema}"
+        auth_issuer = (
+            self.server_config.get("authentication", "server_url")
+        ) + f"/{auth_realm}"
+
+        try:
+            auth_secret = self.server_config.get("authentication", "secret")
+        except Exception:
+            auth_secret = ""
```

Comment:
This means that `secret` is an optional configuration parameter, defaulting to `""`?? And that's OK?